src/Collection/ObjectArrayAccess.php

```diff
@@ -153,20 +153,29 @@ public function __call($name, $arguments)
             return $this;
         }
 
-        // get : fluent setter
+        // set : fluent setter
         if (strpos($name, "set") === 0) {
             $property_name = $this->methodNameToPropertyName(substr($name, 3));
             $this->offsetSet($property_name, $arguments[0]);
             return $this;
         }
 
-        if (strpos($name, "get") === 0) {
+        // get & has : getter
+        if (strpos($name, "get") === 0 || strpos($name, "has") === 0) {
             $property_name = $this->methodNameToPropertyName(substr($name, 3), true);
             if (property_exists($this, $property_name) || $this->__isset($property_name)) {
                 return $this->$property_name;
             }
         }
 
+		// is : getter
+        if (strpos($name, "is") === 0 && strlen($name) > 2 && strtoupper($name[2]) === $name[2]) {
+            $property_name = $this->methodNameToPropertyName(substr($name, 2), true);
+            if (property_exists($this, $property_name) || $this->__isset($property_name)) {
```

Comment:
Est-ce qu'on rajoute une condition && is_bool($this->$property_name) dans le test (avec les parenthèses qui vont bien) pour forcer le type sur le getter "is" ?